### DISCUSSION TITLE
Create new Address object in Reach

### DISCUSSION THREAD

**Discussion by ThangBK2009** [reach-sh/reach-lang#1301] [General]
I want to hardcode an Address in contract to transfer service fee to that. I didn't find any example guide to create Address object from address string (e.g. ZO4YA3FVUCTHEFIMW5V45IJBRTI5T2JBQUPBTD6RP3LVZL2NMG3RIQP6F4)

**Comment by jeapostrophe** [comment]
Here is a link to a prior discussion about this --- https://github.com/reach-sh/reach-lang/issues/434#issuecomment-987853577

The punch-line:
- Adding the syntax to write literal addresses in Reach programs does not add anything you do not already have with `publish`-ing the address in the first message.
- In fact, it makes your program more brittle, less reliable, less usable, and less testable.
- This "feature" is missing on purpose for those reasons.
- Yet, you can still implement exactly the consumer experience you want

**Comment by ThangBK2009** [comment]
Can you guide me how to transfer balance to an external address (it isn't participant in contract)

**Reply by jeapostrophe** [reply]
You can use any `Address` as the `who` argument in `transfer(amt).to(who)`. For example, you can do...
```
Bob.only(() => { const who = declassify(interact.who); });
Bob.publish(who);
transfer(5).to(who);
commit();
```

**Reply by ThangBK2009** [reply]
My use case: All user join to contract must be paid for service fee. This fee will be transfer to a Reward Wallet, this wallet is not change  and I want to hard code in Contract.

**Reply by ThangBK2009** [reply]
In TEAL, I can transfer to any address. But in Reach I don't know do it.

**Reply by jeapostrophe** [reply]
In Reach, you can also transfer to any address. You just give the address to contact (via a publish or API call) and it will be bound to a variable, eg x. Then you use transfer with the to set to that variable.

**Reply by ThangBK2009** [reply]
If I design to let give Reward Address from outside, In case attacker fake my client, he will pass his Wallet to contract as Reward Wallet, then he will not be charged the service fee. So I want to hard code Reward Wallet in the contract.

**Reply by jeapostrophe** [reply]
As I said in the discussion I linked, that doesn't actually help, because you need to verify the contracts that you interact with all the time. You should have anyone who is going to use the contract look at the reward address (through a view, perhaps) and then stop interacting with ones you don't trust.

**Reply by ThangBK2009** [reply]
I am doing a contract to sell NFT, with Seller and Buyer participating in the contract. Do you think there will be another person with the name Reward Wallet also participating in the contract?

**Reply by jeapostrophe** [reply]
No, whoever interacts first (probably the seller) will share the reward address
